### PR TITLE
graphs/matching_covered_graph: override methods concerning loops

### PR DESCRIPTION
**Pull request by janmenjayap**
The objective of this issue is to overwrite the methods pertaining to loops for a matching covered graph. 

More specifically, this PR aims to implement/ update the following methods:
- [x] `allow_loops()`: Change whether loops are allowed in (matching covered) graphs.
- [x] `allows_loops()`: Return whether loops are permitted in (matching covered) graphs.
- [x] `has_loops()`: Check whether thare are loops in the (matching covered) graph.
- [x] `loop_edges()`: Return a list of all loops in the (matching covered) graph.
- [x] `loop_vertices()`: Return a list of vertices with loops.
- [x] `loops()`: Return a list of all loops in the (matching covered) graph.
- [x] `number_of_loops()`: Return the number of edges that are loops.
- [x] `remove_loops()`: Remove loops on vertices in ``vertices``.

This PR shall address the methods related to loops for matching covered graphs.

Fixes #38216. Note that this issue fixes a small part of the mentioned issue.

### :memo: Checklist

- [x] The title is concise and informative.
- [x] The description explains in detail what this PR is about.
- [x] I have linked a relevant issue or discussion.
- [x] I have created tests covering the changes.
- [x] I have updated the documentation and checked the documentation preview.

### :hourglass: Dependencies
This PR depends on the PR #38742.

cc: @dcoudert.

### CODE REVIEW

File: src/sage/graphs/matching_covered_graph.py [hunk 1]
```diff
@@ -0,0 +1,2558 @@
...
+        .. SEEALSO::
+
+            :meth:`~sage.graphs.matching_covered_graph.MatchingCoveredGraph.allow_loops`,
+            :meth:`~sage.graphs.matching_covered_graph.MatchingCoveredGraph.allows_loops`,
+            :meth:`~sage.graphs.matching_covered_graph.MatchingCoveredGraph.loop_edges`,
+            :meth:`~sage.graphs.matching_covered_graph.MatchingCoveredGraph.loop_vertices`,
+            :meth:`~sage.graphs.matching_covered_graph.MatchingCoveredGraph.loops`,
+            :meth:`~sage.graphs.matching_covered_graph.MatchingCoveredGraph.number_of_loops`,
+            :meth:`~sage.graphs.matching_covered_graph.MatchingCoveredGraph.remove_loops`
+
```
Comment (dcoudert) [inline src/sage/graphs/matching_covered_graph.py]: remove empty line

File: src/sage/graphs/matching_covered_graph.py [hunk 2]
```diff
@@ -0,0 +1,2558 @@
...
+            :meth:`~sage.graphs.matching_covered_graph.MatchingCoveredGraph.loop_vertices`,
+            :meth:`~sage.graphs.matching_covered_graph.MatchingCoveredGraph.loops`,
+            :meth:`~sage.graphs.matching_covered_graph.MatchingCoveredGraph.number_of_loops`,
+            :meth:`~sage.graphs.matching_covered_graph.MatchingCoveredGraph.remove_loops`
+        """
+        return []
+
+    @doc_index('Overwritten methods')
+    def loop_vertices(self):
+        """
```
Comment (dcoudert) [inline src/sage/graphs/matching_covered_graph.py]: use `r"""`

**Review by dcoudert — CHANGES_REQUESTED** [review]
I think you added too many doctests. Do we really need to tests all the methods in the tests of all methods ?

**Comment by janmenjayap** [comment]
> I think you added too many doctests. Do we really need to tests all the methods in the tests of all methods ?

Oh.
I will rectify this.

**Review by dcoudert — APPROVED** [review]
LGTM.

**Comment by janmenjayap** [comment]
Hi,
If I am not mistaken, shouldn't we wait till the PR #38742 is merged followed by resolving the conflicts in this PR after pulling the changes from develop branch?

**Comment by dcoudert** [comment]
Yes, it would be much better. I'm setting back this PR to needs review then.

**Review by dcoudert — APPROVED** [review]
LGTM.

**Pull request merged**